cape-char.el

```diff
@@ -78,28 +89,28 @@ PREFIX are the prefix characters."
         (properties (intern (format "cape--%s-properties" name)))
         (thing-re (concat (regexp-opt (mapcar #'char-to-string prefix)) "[^ \n\t]*" )))
     `(progn
-       (defvar ,hash (cape-char--translation
+       (defvar ,hash (cape-char--translation-hash
```

Comment:
This function call should then go into the macro itself with the result let-bound just below `thing-re`. This relates to the comment I've made above about the `eval-when-compile` and purity declare.

Reply:
Sorry I don't fully understand. Is this what you have in mind?

```diff
-  (let ((capf (intern (format "cape-%s" name)))
+  (let* ((capf (intern (format "cape-%s" name)))
         (prefix-required (intern (format "cape-%s-prefix-required" name)))
-        (hash (intern (format "cape--%s-hash" name)))
         (ann (intern (format "cape--%s-annotation" name)))
         (docsig (intern (format "cape--%s-docsig" name)))
         (exit (intern (format "cape--%s-exit" name)))
         (properties (intern (format "cape--%s-properties" name)))
-        (thing-re (concat (regexp-opt (mapcar #'char-to-string prefix)) "[^ \n\t]*" )))
+        (thing-re (concat (regexp-opt (mapcar #'char-to-string prefix)) "[^ \n\t]*" ))
+        (hash (cape-char--translation-hash
+               method
+               (concat "\\`" (regexp-opt (mapcar #'char-to-string prefix))))))
     `(progn
-       (defvar ,hash (cape-char--translation-hash
-                      ,method
-                      ,(concat "\\`" (regexp-opt (mapcar #'char-to-string prefix)))))
```

It works but the hash wouldn't be publicly defined, and references to it later on in other functions in`progn` would be referencing a fully expanded form of the hash, right?

```elisp
(when-let (str (cape-char--ensure-str (gethash name ,hash)))
```

wouldn't this become:

```elisp
(when-let (str (cape-char--ensure-str (gethash name (... hash contents ...))))
```

Which is extremely long?

Reply:
Continued in: https://github.com/minad/cape/pull/91#issuecomment-1715195190